src/Stratis.SmartContracts.CLR/Serialization/ContractPrimitiveSerializer.cs

```diff
@@ -155,42 +144,59 @@ public object Deserialize(Type type, byte[] stream)
             if (stream == null || stream.Length == 0)
                 return null;
 
-            if (type == typeof(byte[]))
-                return stream;
+            switch (Type.GetTypeCode(type))
```

Comment:
I recommends switch expression syntax 

```
            return Type.GetTypeCode(type) switch
            {
                TypeCode.Byte => stream[0],
                TypeCode.Char => ToChar(stream),
                TypeCode.Boolean => ToBool(stream),
                TypeCode.Int32 => ToInt32(stream),
                TypeCode.Int64 => ToInt64(stream),
                TypeCode.String => ToString(stream),
                TypeCode.UInt32 => ToUInt32(stream),
                TypeCode.UInt64 => ToUInt64(stream),
                _ when type == typeof(UInt128) => ToUInt128(stream),
                _ when type == typeof(uint256) => ToUInt256(stream),
                _ when type == typeof(Address) => ToAddress(stream),
                _ when type.IsArray => DeserializeArray(type.GetElementType(), stream),
                _ when type.IsValueType => DeserializeStruct(type, stream),
                _ => throw new ContractPrimitiveSerializationException(string.Format("{0} is not supported.", type.Name)),
            };
```

Reply:
This is unrelated to this PR so normally I would not make this change. Let me add it anyway since you did invest some time.